src/controllers/SecretaryLoadedPatientController.java

```diff
@@ -150,27 +144,74 @@ private Command RescheduleAppointment() {
             } else {
                 LocalDate day = secretaryScreenView.bookAppointmentDayPrompt();
                 AppointmentTimeDetails time = secretaryScreenView.bookAppointmentTimePrompt();
-                appointmentManager.rescheduleAppointment(
-                        appointments.get(index), day.getYear(), day.getMonthValue(),
-                        day.getDayOfMonth(), time.time().getHour(), time.time().getMinute(), time.length());
+                LocalDateTime startTime = LocalDateTime.of(day.getYear(), day.getMonthValue(), day.getDayOfMonth(),
```

Comment:
What happens when day is null? This code smells like null pointer errors. What happens when time is null? Again null pointer errors. Furthermore, if the call to  `getPatientAppointments` returns an empty list a relevant presenter method NoUserAppointmentsMessage should be displayed and the command should not continue.